src/components/c-google-maps.vue

```diff
@@ -0,0 +1,372 @@
+<template>
+  <div :class="b()">
+    <div
+      ref="container"
+      :class="b('container')"
+    ></div>
+  </div>
+</template>
+
+<script>
+  import { mapGetters } from 'vuex';
+  import store from '@/store/index';
+  import loadScript from '@/helpers/load-script';
+  import { GOOGLE_MAPS_THEME_GRAY } from '@/setup/globals';
+
+  let geocoder;
+
+  /**
+   * Renders a Google Map.
+   */
+  export default {
+    name: 'c-google-maps',
+
+    // components: {},
+    // mixins: [],
+
+    props: {
+      /**
+       * Expects a list of locations (markers). You can define additional properties if required.
+       *
+       * lat/lng OR geocode MUST be defined!
+       *
+       * @property {Number} [lat] - The latitude of the location.
+       * @property {Number} [lng] - The longitude of the location.
+       * @property {String} [geocode] - An address string, that should be used to geocode lat/lng if they are missing.
+       * @property {String} [icon] - A specific icon source for the location.
+       * @property {String} [title] - A title for the marker icon.
+       */
+      locations: {
+        type: Array,
+        required: true,
+      },
+
+      /**
+       * Allows to use an alternative map theme.
+       *
+       * Valid values: `[default, gray]`
+       */
+      theme: {
+        type: String,
+        default: 'default',
+        validator(value) {
+          return [
+            'default',
+            'gray',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Allows to set a custom bounds or en-/disable the auto bounding.
+       *
+       * @see https://developers.google.com/maps/documentation/javascript/reference/coordinates#LatLngBounds
+       */
+      bounds: {
+        type: [Object, Boolean],
```

Comment:
todo: please change the order of types. Else this can lead to unexpected results if the attribute is defined without value. See https://eslint.vuejs.org/rules/prefer-true-attribute-shorthand.html for more information.

`[Boolean, Object]`